### DISCUSSION TITLE
A large number of idle steps (bug or feature)

### DISCUSSION THREAD

**Discussion by websolutions-hamburg** [vitest-dev/vitest#2916] [Q&A] [answered]
Hello,

I'm not sure if this is a bug or more a expected behavior. When I analyse the vitest run using `--inspect-brk`, I see a lot of idle steps. Is it supposed to be like that? Couldn't vitest be much faster if it didn't have these idle steps or if they were much shorter?

I testet it with: `NODE_OPTIONS="--inspect-brk" vitest run`.
The cli option `vitest run --inspect-brk` was not working. Is this a bug or do I something wrong?

![image](https://user-images.githubusercontent.com/26099903/221223191-d9ed7c44-1b8f-4b2f-958b-1fbf1f666aad.png)

**Comment by sheremet-va** [comment, marked as answer]
@Aslemammad what do you think?

**Reply by sheremet-va** [reply]
I think this might be just workers running? Should they appear with `--inspect-brk` or do we need to pass something there?

**Reply by sheremet-va** [reply]
Looks like is indeed worker threads running: https://bugs.chromium.org/p/chromium/issues/detail?id=968472

**Reply by Aslemammad** [reply]
As far as I know dev tools do not support workers, so that maybe it.

**Reply by AriPerkkio** [reply]
`--inspect-brk` won't include the worker threads but you can see their execution using `--cpu-prof`. The main thread and each worker thread will generate their own results. Those idle ones are indeed the worker threads. 

```bash
$ node --cpu-prof --cpu-prof-dir=./profiling ./node_modules/vitest/vitest.mjs 

$ ls profiling/
CPU.20230224.202136.93565.0.001.cpuprofile
CPU.20230224.202137.93565.1.002.cpuprofile
CPU.20230224.202137.93565.2.003.cpuprofile
CPU.20230224.202137.93565.3.004.cpuprofile
CPU.20230224.202137.93565.4.005.cpuprofile
```

You can then open these in Chrome Dev tools:
![image](https://user-images.githubusercontent.com/14806298/221260055-117bddcd-849e-4dd9-9aba-4888bae5291d.png)

I've been looking for a tool that could merge these `.cpuprofile` files but haven't found any. Ideally there would be a single report where each "idle" part would be filled with worker threads equivalent report.

**Reply by websolutions-hamburg** [reply]
Thanks for the good explanation!

**Comment by websolutions-hamburg** [comment]
One more question about the CLI options `--inspect` & `--inspect-brk` which are listed here https://vitest.dev/guide/cli.html.
The way I read it, you should be able to call up the options directly in combination with the `vitest` command.
Like `vitest run --inspect-brk`.
But this does not work for me. Is this a bug or a wrong documentation?
